File: NaSt2D-2.0/Tools/c/revers.c
/*     revers.c */
/********************************************************************/
/*     Program for data output in reverse order                     */
/*     -1. -1. -1. labelled the input end                           */
/*------------------------------------------------------------------*/
/*     Programm zur Datenausgabe in umgekehrter Folge               */
/*     -1. -1. -1. kennzeichnet das Eingabeende                     */
/********************************************************************/
#include <stdio.h>
#include <stdlib.h>

#define VOL1 1000   /* 1st dimension row    - Zeilenindex i  */
#define VOL2 3      /* 2nd dimension column - Spaltenindex j */

int main(void) 
{
   int    i=0,in=0,j;
   float  array[VOL1][VOL2];
   FILE   *fpein;
   FILE   *fpaus;
   char   *input="ein";           /* input file  */
   char   *output="aus";          /* output file */

   fpein=fopen(input,"r");        /* opening the files */
   fpaus=fopen(output,"w");
   if ((fpein==NULL)||(fpaus==NULL)) {
      printf("Error on opening the files\n");
      return(-1);                 /* EXIT_ERROR_input_file */
                                     }

   for(i=0; i < VOL1; i++) {      /* read in array */
      for(j=0; j < VOL2; j++) fscanf(fpein,"%f ", &array[i][j]);
      in=i;
      if((array[i][0] == array[i][1]) && (array[i][0] == array[i][2])) break;
                           }

   for(i=in; i >= 0; i--) {       /* write array in revers order */
      for(j=0; j < VOL2; j++) fprintf(fpaus,"%3.8f ",array[i][j]);
      fprintf(fpaus,"\n");
                          }

   printf("revers EXIT_SUCCESS\n");
   fclose(fpein);
   fclose(fpaus);
   return EXIT_SUCCESS;
}

File: NaSt2D-2.0/Tools/f/revers.f
c********************************************************************
c     Program for data output in reverse order
c     -1. -1. -1. labelled the input end
c--------------------------------------------------------------------
c     Programm zur Datenausgabe in umgekehrter Folge
c     -1. -1. -1. kennzeichnet das Eingabeende
c********************************************************************
      program Revers
c
      parameter (n=10000)      
      dimension a(n),b(n),c(n) 
      open (unit=1,file='ein')
      open (unit=2,file='aus')
c
      do 10 i=1,n      
      read(1,*) a(i),b(i),c(i) 
      if(a(i).eq.-1..and.b(i).eq.-1..and.c(i).eq.-1.) goto 20
   10 continue
   20 in=i
      do 30 i=1,in
      j=in-i+1
      write(2,*) a(j),b(j),c(j)  
   30 continue
      stop 'ok'
      end
c*********************************************************************




















